src/commands/Minion/favpatch.ts

```diff
@@ -0,0 +1,77 @@
+import { objectValues } from 'e';
+import { ArrayActions, CommandStore, KlasaMessage } from 'klasa';
+
+import { FarmingPatchTypes } from '../../lib/minions/farming/types';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { BotCommand } from '../../lib/structures/BotCommand';
+import { toTitleCase } from '../../lib/util';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			usage: '[patches:...string]',
+			description: 'Favorites a farming patch.',
+			examples: ['+favpatch herb'],
+			categoryFlags: ['minion', 'skilling']
+		});
+	}
+
+	async run(msg: KlasaMessage, [patches]: [string | undefined]) {
+		const currentFavorites = msg.author.settings.get(UserSettings.FavoritePatches);
+		let newFavorites = [...currentFavorites];
+		const input = patches?.split(', ').map(x => toTitleCase(x).replace(' ', ''));
+
+		if (!patches || patches.length === 0) {
+			if (currentFavorites.length === 0) {
+				return msg.channel.send(
+					`You have no favorited farming patches. Valid options are: ${objectValues(FarmingPatchTypes)}`
+				);
+			}
+			return msg.channel.send(
+				`Your current favorite farming patches are: ${currentFavorites.toString().split(',').join(', ')}.`
```

Comment:
all the `.toString().split()` in these are really weird, youre turning an array into a string, then back into an array, just remove the `.toString().split()`

Reply:
They're specifically there to convert "patch1,patch2,patch3" into "patch1, patch2, patch3", adding a space after commas. I wanted to avoid regex.